internal/kafka/internal/services/kafka.go

```diff
@@ -619,24 +625,24 @@ func (k *kafkaService) DeprovisionKafkaForUsers(users []string) *errors.ServiceE
 	return nil
 }
 
-func (k *kafkaService) DeprovisionExpiredKafkas() *errors.ServiceError {
-	dbConn := k.connectionFactory.New().Model(&dbapi.KafkaRequest{}).Session(&gorm.Session{})
-
-	var typesWithLifespan []string
-	for _, kafkaInstanceType := range k.kafkaConfig.SupportedInstanceTypes.Configuration.SupportedKafkaInstanceTypes {
-		if kafkaInstanceType.HasAnInstanceSizeWithLifespan() {
-			typesWithLifespan = append(typesWithLifespan, kafkaInstanceType.Id)
-		}
+func (k *kafkaService) kafkaWithExpiresAtShouldBeDeprovisioned(kafkaRequest *dbapi.KafkaRequest, currentTime time.Time) (bool, error) {
+	glog.V(10).Infof("Evaluating expiration time of kafka request '%s' with instance type '%s', size ID '%s' and status '%s'", kafkaRequest.ID, kafkaRequest.InstanceType, kafkaRequest.SizeId, kafkaRequest.Status)
+	if currentTime.After(kafkaRequest.ExpiresAt.Time) {
```

Comment:
I believe we should check if `ExpiresAt` is `nil`.

I think we can use a method like this here: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/be6dc11b91d2a4d4e0015c9b623e8febe3dc0d47/internal/kafka/internal/api/dbapi/kafka_request_types.go#L145

Reply:
There is no need to check it. This method is a private method and it is only called when the time is a valid time. It assumes a precondition of the time being valid